File: go/models/asset.go
package models

import (
	"time"
)

// Owner: dispatch-asset-service
// Consumers: backend, mission-command-service, location-navigation-service, field-agent-app

// Asset status constants - simplified to 4 logical statuses
const (
	AssetStatusAvailable  = "available"  // Ready for dispatch
	AssetStatusDispatched = "dispatched" // On a mission
	AssetStatusReturning  = "returning"  // Coming back to base
	AssetStatusOffline    = "offline"    // Not operational
)

// Asset represents an asset in the system
type Asset struct {
	ID               string                 `json:"id"`
	Name             string                 `json:"name"`
	Type             string                 `json:"type"`
	Status           string                 `json:"status"`
	UseCase          string                 `json:"useCase"`
	TeamID           string                 `json:"teamId,omitempty"`      // Team this asset belongs to
	AssignedAreaIds  []string               `json:"assignedAreaIds,omitempty"` // Areas this asset patrols/monitors
	Latitude         float64                `json:"latitude"`
	Longitude        float64                `json:"longitude"`
	Altitude         float64                `json:"altitude,omitempty"`
	BatteryLevel     int                    `json:"batteryLevel,omitempty"`
	Members          int                    `json:"members,omitempty"`
	Vehicle          string                 `json:"vehicle,omitempty"`
	PulseRate        int                    `json:"pulseRate,omitempty"`
	OxygenLevel      int                    `json:"oxygenLevel,omitempty"`
	Location         string                 `json:"location,omitempty"`
	DispatchTime     *time.Time             `json:"dispatchTime,omitempty"`
	EstimatedArrival *time.Time             `json:"estimatedArrival,omitempty"`
	LastUpdated      time.Time              `json:"lastUpdated"`
	LastVitalUpdate  *time.Time             `json:"lastVitalUpdate,omitempty"`
	VideoSrc            string                 `json:"videoSrc,omitempty"`
	Metadata            map[string]interface{} `json:"metadata,omitempty"`
	AutoPositionEnabled bool                   `json:"autoPositionEnabled"` // When false, position simulator skips this asset
}

File: typescript/src/models/Asset.ts
/**
 * Asset model
 * Converted from go/models/asset.go
 * Owner: dispatch-asset-service
 * Consumers: backend, mission-command-service, location-navigation-service, field-agent-app
 */

/** Asset status constants - simplified to 4 logical statuses */
export const AssetStatus = {
  AVAILABLE: 'available' as const,  // Ready for dispatch
  DISPATCHED: 'dispatched' as const, // On a mission
  RETURNING: 'returning' as const,   // Coming back to base
  OFFLINE: 'offline' as const,       // Not operational
};

export type AssetStatusType = typeof AssetStatus[keyof typeof AssetStatus];

/** Asset represents an asset in the system */
export interface Asset {
  id: string;
  name: string;
  type: string;
  status: string;
  useCase: string;
  teamId?: string; // Team this asset belongs to
  assignedAreaIds?: string[]; // Areas this asset patrols/monitors
  latitude: number;
  longitude: number;
  altitude?: number;
  batteryLevel?: number;
  members?: number;
  vehicle?: string;
  pulseRate?: number;
  oxygenLevel?: number;
  location?: string;
  dispatchTime?: string; // ISO 8601
  estimatedArrival?: string; // ISO 8601
  lastUpdated: string; // ISO 8601
  lastVitalUpdate?: string; // ISO 8601
  videoSrc?: string;
  metadata?: Record<string, any>;
  autoPositionEnabled: boolean; // When false, position simulator skips this asset
}
